rust/kernel/error.rs

```diff
@@ -57,14 +57,34 @@ impl Error {
     pub const EBADF: Self = Error(-(bindings::EBADF as i32));
 
     /// Creates an [`Error`] from a kernel error code.
+    /// when errno given is invalid,
+    /// it will print a warning message and convert it to EINVAL
     pub fn from_kernel_errno(errno: c_types::c_int) -> Error {
+        Error::new(errno)
+    }
+
+    /// Creates an [`Error`] from a kernel error code without a sanity check
+    pub fn from_kernel_errno_unchecked(errno: c_types::c_int) -> Error {
         Error(errno)
     }
 
     /// Returns the kernel error code.
     pub fn to_kernel_errno(self) -> c_types::c_int {
         self.0
     }
+
+    /// INVARIANT: make sure Error is initialized with a sane value
+    /// When an invalid errno is found, it will
+    /// 1) print a warning message
+    /// 2) convert it to EINVAL
+    pub fn new(errno: c_types::c_int) -> Error {
+        if errno < -(bindings::MAX_ERRNO as i32) || errno >= 0 {
+            crate::pr_warn!("Creating Error with an invalid errno {}, convert it to EINVAL", errno);
```

Comment:
I guess this is an OK approach to avoid a panic.

Reply:
There is a caveat. `printk` will grab the console_lock. This is generally ok, but when one is writing a console-related thing that may lead to a potential deadlock. So I think we surely need an unchecked version from_kernel_errno_unchecked.

Reply:
I was mistaken. I've checked printk code. If it failed to grab console_lock, it would use another approach. It won't be a problem.